lib/ecto_diff.ex

```diff
@@ -428,7 +468,9 @@ defmodule EctoDiff do
     keys
   end
 
-  defp no_changes?(%{effect: :changed, changes: map}) when map == %{}, do: true
+  defp no_changes?([_ | _] = changes), do: Enum.any?(changes, &no_changes?/1)
+  defp no_changes?([]), do: true
+  defp no_changes?(%{effect: effect, changes: map}) when map == %{} and effect in [:added, :changed], do: true
```

Comment:
Can we add a small comment at the top of this function explaining the logic behind this?

Reply:
no longer relevant